Dawarich update from 1.6.1 to  1.7.0

### ✅ Have you read and understood the above guidelines?

yes

### 🔎 Did you run the script with verbose mode enabled?

Yes, verbose mode was enabled and the output is included below

### 📜 What is the name of the script you are using?

dawarich

### 📂 What was the exact command used to execute the script?

update

### ⚙️ What settings are you using?

- [x] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 13

### 📈 Which Proxmox version are you on?

9.1.9

### 📝 Provide a clear and concise description of the issue.

(See full trace by running task with --trace)
  ✖️   in line 70: exit code 1 (General error / Operation not permitted): while executing command  bundle exec rake assets:precompile

### 🔄 Steps to reproduce the issue.

.

### ❌ Paste the full error output (if available).

root@dawarich:~# cat /tmp/create-lxc-3628ffd6.log
[2026-04-27 06:03:38] [INFO] Checking for update: dawarich
[2026-04-27 06:03:39] [OK] Update available: dawarich 1.6.1 → 1.7.0
[2026-04-27 06:03:39] [INFO] Stopping Services
[2026-04-27 06:03:41] [OK] Stopped Services
[2026-04-27 06:03:41] [INFO] Backing up Data
[2026-04-27 06:03:41] [OK] Backed up Data
[2026-04-27 06:03:42] [INFO] Fetching GitHub release: dawarich (1.7.0)
[2026-04-27 06:03:46] [OK] Deployed: dawarich (1.7.0)
[2026-04-27 06:03:46] [INFO] Update Ruby 3.4.6
[2026-04-27 06:03:46] [OK] Update Ruby 3.4.6
[2026-04-27 06:03:46] [INFO] Running Migrations
[2026-04-27 06:05:00] [ERROR] in line 70: exit code 1 (General error / Operation not permitted): while executing command  bundle exec rake assets:precompile
[2026-04-27 06:05:00] Full log: /tmp/create-lxc-3628ffd6.log
root@dawarich:~# 

### 🖼️ Additional context (optional).

_No response_

--- a/ct/dawarich.sh
+++ b/ct/dawarich.sh
@@ -53,6 +53,10 @@ function update_script() {
     export PATH="/root/.rbenv/shims:/root/.rbenv/bin:$PATH"
     eval "$(/root/.rbenv/bin/rbenv init - bash)"
 
+    if ! grep -q "OTP_ENCRYPTION_PRIMARY_KEY" /opt/dawarich/.env; then
+      echo "OTP_ENCRYPTION_PRIMARY_KEY=$(openssl rand -hex 32)" >>/opt/dawarich/.env
+    fi
+
     set -a && source /opt/dawarich/.env && set +a
 
     $STD bundle config set --local deployment 'true'

--- a/ct/dawarich.sh
+++ b/ct/dawarich.sh
@@ -67,8 +67,8 @@ function update_script() {
       $STD npm install
     fi
 
-    $STD bundle exec rake assets:precompile
     $STD bundle exec rails db:migrate
+    $STD bundle exec rake assets:precompile
     $STD bundle exec rake data:migrate
     msg_ok "Ran Migrations"
 

--- a/install/dawarich-install.sh
+++ b/install/dawarich-install.sh
@@ -46,10 +46,12 @@ msg_ok "Set up Directories"
 
 msg_info "Configuring Environment"
 SECRET_KEY_BASE=$(openssl rand -hex 64)
+OTP_ENCRYPTION_PRIMARY_KEY=$(openssl rand -hex 32)
 RELEASE=$(get_latest_github_release "Freika/dawarich")
 cat <<EOF >/opt/dawarich/.env
 RAILS_ENV=production
 SECRET_KEY_BASE=${SECRET_KEY_BASE}
+OTP_ENCRYPTION_PRIMARY_KEY=${OTP_ENCRYPTION_PRIMARY_KEY}
 DATABASE_HOST=localhost
 DATABASE_USERNAME=${PG_DB_USER}
 DATABASE_PASSWORD=${PG_DB_PASS}

